AsyncDisplayKit/Private/ASPendingStateController.mm

```diff
@@ -65,8 +64,8 @@ - (void)flush
 {
   ASDisplayNodeAssertMainThread();
   _lock.lock();
-    ASWeakSet *dirtyNodes = _dirtyNodes;
-    _dirtyNodes = [[ASWeakSet alloc] init];
+    NSArray *dirtyNodes = [_dirtyNodes allObjects];
+    [_dirtyNodes removeAllObjects];
```

Comment:
I'm not actually sure this is faster than letting the object deallocate itself; plus it will create a memory high watermark (not that this is the biggest issue)